File: backend/services/ipinfo_service.py
import requests
import os
from dotenv import load_dotenv

load_dotenv()

class IPInfoService:
    BASE_URL = "https://ipinfo.io"

    def __init__(self):
        self.token = os.getenv("IPINFO_TOKEN")
    
    def get_ip_details(self, ip_address):
        """Fetch geolocation and details for an IP."""
        if not self.token:
            print("IPINFO_TOKEN not set.")
            return None
            
        try:
            url = f"{self.BASE_URL}/{ip_address}"
            params = {"token": self.token}
            response = requests.get(url, params=params)
            response.raise_for_status()
            return response.json()
        except Exception as e:
            print(f"Error fetching IP details for {ip_address}: {e}")
            return None

File: backend/services/ipinfoService.js
const axios = require('axios');
const path = require('path');
require('dotenv').config({ path: path.resolve(__dirname, '../.env') });

const TOKEN = process.env.IPINFO_TOKEN;
const BASE_URL = 'https://ipinfo.io';

const getIpDetails = async (ip) => {
    try {
        const response = await axios.get(`${BASE_URL}/${ip}`, {
            params: {
                token: TOKEN
            }
        });
        return response.data;
    } catch (error) {
        console.error('Error fetching IP details from IPinfo:', error.message);
        return null;
    }
};

module.exports = { getIpDetails };
